org.adempiere.base/src/org/compiere/model/GridTable.java

```diff
@@ -1143,11 +1143,19 @@ public Object getValueAt (int row, int col)
 	 * @param row
 	 */
 	public void waitLoadingForRow(int row) {
+		// optimization - this method is called too frequently
+		// avoid processing when the criteria of whiles and if's below is not  met
+		if ( ! (row >= m_sort.size() || m_sort.size() == 0) )
+			return;
 		//	need to wait for data read into buffer
 		int loops = 0;
 		//wait for [timeout] seconds
 		int timeout = MSysConfig.getIntValue(MSysConfig.GRIDTABLE_LOAD_TIMEOUT_IN_SECONDS, DEFAULT_GRIDTABLE_LOAD_TIMEOUT_IN_SECONDS, Env.getAD_Client_ID(Env.getCtx()));
-		while (row >= m_sort.size() && m_loaderFuture != null && !m_loaderFuture.isDone() && loops < timeout)
+		if (timeout <= 0)
+			timeout = 120; /* wait max 2 minutes if the load timeout is not set, WARNING, this is not recommended as the query keeps running in the database */

```

Comment:
If it is a "warning", perhaps we should log it as a warning message. If we log this, would be better if we don't log the same warning > 1 times (I think we don't have such support now).

Reply:
> If it is a "warning", perhaps we should log it as a warning message. If we log this, would be better if we don't log the same warning > 1 times (I think we don't have such support now).

done, I think is OK to flood the log with these warnings as this becomes a really bad practice that can potentially crash the server